typescript/infra/scripts/agent-utils.ts

```diff
@@ -122,39 +131,100 @@ export function assertEnvironment(env: string): DeployEnvironment {
   );
 }
 
-export function getEnvironmentConfig(environment: DeployEnvironment) {
-  return environments[environment];
-}
-
-export async function getConfigsBasedOnArgs(argv?: {
+// not requiring to build coreConfig to get agentConfig
+export async function getAgentConfigsBasedOnArgs(argv?: {
   environment: DeployEnvironment;
   context: Contexts;
+  'new-chains': string;
 }) {
-  const { environment, context = Contexts.Hyperlane } = argv
-    ? argv
-    : await withContext(getArgs()).argv;
-  const envConfig = getEnvironmentConfig(environment);
-  const agentConfig = getAgentConfig(context, envConfig);
-  return { envConfig, agentConfig, context, environment };
+  const {
+    environment,
+    context = Contexts.Hyperlane,
+    'new-chains': newChains,
+  } = argv ? argv : await withMissingChains(withContext(getArgs())).argv;
+
+  const newValidatorCounts: ChainMap<number> = {};
+  const newThresholds: ChainMap<number> = {};
+  if (newChains) {
+    const chains = newChains.split(',');
+    for (const chain of chains) {
+      const [chainName, threshold] = chain.split('=');
+      const [newThreshold, newValidatorCount] = threshold.split('/');
+      newThresholds[chainName] = Number(newThreshold);
+      newValidatorCounts[chainName] = Number(newValidatorCount);
```

Comment:
maybe parseInt?

Reply:
ngl this feels pretty fragile
idk if a regex is better?

Reply:
This is similar to the parseContextAndRoles for the fund-keys-from-deployer script